apps/ocpp/store/redis_state.py

```
@@ -0,0 +1,122 @@
+    key = _redis_ip_key(ip)
+    token = _connection_token(consumer)
+    try:
+        pipe = client.pipeline()
+        pipe.sadd(key, token)
+        pipe.expire(key, _IP_CONNECTION_TTL)
+        pipe.scard(key)
+        added, _expired, count = pipe.execute()
+        if count > MAX_CONNECTIONS_PER_IP and added:
+            client.srem(key, token)
+            return False
+        return count <= MAX_CONNECTIONS_PER_IP
```

Ruff (0.14.14)

[warning] Consider moving this statement to an `else` block

(TRY300)